src/app/features/rpc-transaction-request/use-rpc-transaction-request.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from 'react';
+
+import type { RpcMethodNames } from '@leather.io/rpc';
+
+import { RpcErrorMessage } from '@shared/rpc/methods/validation.utils';
+import { closeWindow } from '@shared/utils';
+import { analytics } from '@shared/utils/analytics';
+
+import { focusTabAndWindow } from '@app/common/focus-tab';
+import { useRpcRequestParams } from '@app/common/hooks/use-rpc-request-params';
+
+export interface RpcTransactionRequest {
+  origin: string;
+  requestId: string;
+  tabId: number;
+  isBroadcasting: boolean;
+  isLoading: boolean;
+  isSubmitted: boolean;
+  onSetIsBroadcasting(value: boolean): void;
+  onSetIsLoading(value: boolean): void;
+  onSetIsSubmitted(value: boolean): void;
+}
+
+export function useRpcTransactionRequest() {
+  const [isBroadcasting, setIsBroadcasting] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSubmitted, setIsSubmitted] = useState(false);
```

Comment:
I wonder if `useReducer` might be a nicer pattern to use here, as it can be a little bit more like a state machine, e.g. preventing illegal states like isBroadcasting true, isSubmitted false.

Reply:
Good idea, I'll prob merge this but then tackle that in refactoring `stx_transferSip10Ft` next.